### ISSUE TITLE
[bitnami/airflow] LDAP error - rolesMapping is not working

### ISSUE THREAD

**Issue by zibuyule** [bitnami/charts#20374]
### Name and Version

bitnami/airflow-16.0.6

### What architecture are you using?

amd64

### What steps will reproduce the bug?

When I start airflow with the following configuration, I log in to the UI interface of airflow. I will get the following error:
Your user has no roles and/or permissions!

When I set the value of userRegistrationRole to User, the user's role after logging in is User; when I set the value of userRegistrationRole to Admin, the user's role after logging in is Admin.

And my user tom belongs to the test group. No matter I configure it as "cn=test,ou=Groups,dc=company,dc=ai": ["User"] or "cn=test,ou=Groups,dc=company,dc=ai": ["Admin"], it all looks like it's not working. The permissions of a user's login depend on the value of userRegistrationRole.

### Are you using any custom parameters or values?

Posted here is part of my values.yaml value

```yaml
ldap:
  enabled: true
  uri: "ldap://dc.domain.com:389"
  basedn: "DC=domain,DC=ai"
  searchAttribute: "cn"
  binddn: "CN=ldapservice,OU=China,OU=Users,OU=company, DC=domain,DC=ai"
  bindpw: "password"
  userRegistration: 'True'
  userRegistrationRole: "Public"
  rolesMapping: '{ "cn=test,ou=Groups,dc=company,dc=ai": ["Admin"], "cn=Admins,ou=Groups,dc=company,dc=ai": ["User"], }'
  rolesSyncAtLogin: 'True'
```

My AD domain user tom belongs to group test.

When I started airflow with the above configuration, I saw the following page:
![no-roles](https://github.com/bitnami/charts/assets/5792519/36723b08-918b-4bdb-9aa6-20b0e124af1a)

### What is the expected behavior?

I hope to assign the user tom an admin or USer permission through rolesMapping.

**Comment by carrodher** [comment]
It appears that the issue may not be directly related to the Bitnami Airflow container image or Helm chart, but rather to how the application is being utilized or configured in your specific environment.

For inquiries about the application itself, customizing its content, or questions about technology and infrastructure usage, we highly recommend referring to the forums and user guides provided by the project responsible for the application or technology.

With that said, we'll keep this ticket open until the stale bot automatically closes it, in case someone from the community contributes valuable insights.

**Comment by zibuyule** [comment]
Sorry, I don't agree with your statement.

These are the parameters for ldap provided by bitnami/airflow. I have spent three days repeatedly trying various configurations and combinations. Either rolesMapping does not provide the functionality I expect, or this is a bug.


My expectation is that users who belong to the SzManagement group in the AD domain, such as tom, are mapped to the Admin role in airflow.


But no matter how I set rolesMapping, it is invalid.
The role obtained by tom actually depends on the value of userRegistrationRole.
If userRegistrationRole is set to Admin, then tom will log in to the airflow UI page and obtain the Admin role.
